binary_search_tree/tree.py

```diff
@@ -14,45 +16,148 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def add(self, key, value = None):
-        pass
+        tree = TreeNode(key, value)
+
+        if self.root == None: 
+            self.root = tree
+        else: 
+            current = self.root 
+
+            while current != None: 
+                if key == current.key: 
+                    return None
+                if key < current.key: 
+                    if current.left == None:
+                        current.left = tree
+                    current = current.left 
+                elif key > current.key:
+                    if current.right == None:
+                        current.right = tree
+                    current = current.right 
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def find(self, key):
```

Comment:
👍  Time complexity is O(n) if the tree is unbalanced and O(log n) if it is balanced.